modules/percolator/src/main/java/org/elasticsearch/percolator/QueryAnalyzer.java

```diff
@@ -619,11 +637,20 @@ private static BytesRef smallestRange(Set<QueryExtraction> terms) {
         final Set<QueryExtraction> extractions;
         final boolean verified;
         final int minimumShouldMatch;
+        final boolean matchAllDocs;
 
         Result(boolean verified, Set<QueryExtraction> extractions, int minimumShouldMatch) {
             this.extractions = extractions;
             this.verified = verified;
             this.minimumShouldMatch = minimumShouldMatch;
+            this.matchAllDocs = false;
+        }
+
+        Result(boolean matchAllDocs, boolean verified, Set<QueryExtraction> extractions, int minimumShouldMatch) {
+            this.extractions = extractions;
```

Comment:
Maybe this constructor should only take `boolean matchAllDocs` as a parameter? If all docs match, then verified is implicitly true, extractions should be empty and minimumShouldMatch should be 0?

Reply:
We do need to take `verified` into account. Lets say there is a bool with a match_all as must and a must_not clause, then that can never be verified.